packages/docusaurus-theme-classic/src/theme/Tabs/index.tsx

```diff
@@ -48,6 +54,27 @@ function Tabs(props: Props): JSX.Element {
 
     if (groupId != null) {
       setTabGroupChoices(groupId, selectedTabValue);
+
+      setTimeout(() => {
+        if (domUtils.isInViewport(selectedTab)) {
+          return;
+        }
+
+        selectedTab.scrollIntoView({
+          block: 'start',
+        });
+
+        setTimeout(() => {
+          if (hideOnScroll) {
+            setNavbarVisible(false);
+          } else {
+            const navbarHeight = domUtils.convertRemToPx(
+              domUtils.getPropertyValue('--ifm-navbar-height'),
+            );
+            window.scrollBy(0, -navbarHeight);
+          }
+        }, 50);
+      }, 150);
```

Comment:
In an ideal world we'd want the mouse cursor to stay above the tab we just clicked without the user seeing things scroll (using useLayoutEffect), but this is more complex to achieve.

As your solution is not the ideal solution, but more a "best effort, good enough", and I think it's a bit too complicated for little value.

Something like this is simpler and even better in my opinion:

```
    if (groupId != null) {
      setTabGroupChoices(groupId, selectedTabValue);

      setTimeout(() => {
        if (domUtils.isInViewport(selectedTab)) {
          return;
        }

        selectedTab.scrollIntoView({
          block: 'center',
          behavior: 'smooth',
        });
      }, 150);
    }
```

block start is not the best for me because the user will click on a tab in the middle of the screen and that tab will appear at the very top of the screen afterward. Without animation, he may struggle to understand what is happening. Centering the tab makes more sense to me and handle the collapsible scrollbar at the same time with less complexity and much less code to ship/maintain

Reply:
Alright, so I also added a short blicker on the active tab item to point the user to the correct tab. Or do you think this is superfluous?

Changed behavior:

![image](https://user-images.githubusercontent.com/4408379/108382199-9c7b3080-7219-11eb-8f13-1268c1927740.png)

I have not reverted the other code. For example, managing the state of the hideable navbar is useful to us to fix a bug with docsearch (after closing the search modal, the navbar is closed for some reason).